ocs_ci/utility/rosa.py

```diff
@@ -0,0 +1,224 @@
+# -*- coding: utf8 -*-
+"""
+Module for interactions with Openshift Dedciated Cluster.
+"""
+
+
+import json
+import logging
+import os
+import re
+import time
+
+from ocs_ci.framework import config
+from ocs_ci.ocs.exceptions import ManagedServiceAddonDeploymentError
+from ocs_ci.utility import openshift_dedicated as ocm
+from ocs_ci.utility import utils
+from ocs_ci.utility.version import get_semantic_version
+
+logger = logging.getLogger(name=__file__)
+rosa = config.AUTH.get("rosa", {})
+
+
+def login():
+    """
+    Login to ROSA client
+    """
+    token = ocm["token"]
+    cmd = f"rosa login --token={token}"
+    logger.info("Logging in to ROSA cli")
+    utils.run_cmd(cmd, secrets=[token])
+    logger.info("Successfully logged in to ROSA")
+
+
+def create_cluster(cluster_name, version):
+    """
+    Create OCP cluster.
+
+    Args:
+        cluster_name (str): Cluster name.
+        version (str): cluster version
+
+    """
+    create_account_roles(version)
+    region = config.DEPLOYMENT["region"]
+    compute_nodes = config.ENV_DATA["worker_replicas"]
+    compute_machine_type = config.ENV_DATA["worker_instance_type"]
+    cmd = (
+        f"rosa create cluster --cluster-name {cluster_name} --region {region} "
+        f"--compute-nodes {compute_nodes} --compute-machine-type "
+        f"{compute_machine_type}  --version {version} --sts --yes"
+    )
+    utils.run_cmd(cmd)
+    create_operator_roles(cluster_name)
+    create_oidc_provider(cluster_name)
+    logger.info("Waiting for installation of ROSA cluster")
+    for cluster_info in utils.TimeoutSampler(
+        10000, 30, ocm.get_cluster_details, cluster_name
+    ):
+        status = cluster_info["status"]["state"]
+        logger.info(f"Current installation status: {status}")
+        if status == "ready":
+            logger.info("Cluster was installed")
+            break
+    cluster_info = ocm.get_cluster_details(cluster_name)
+    # Create metadata file to store the cluster name
+    cluster_info["clusterName"] = cluster_name
+    cluster_info["clusterID"] = cluster_info["id"]
+    cluster_path = config.ENV_DATA["cluster_path"]
+    metadata_file = os.path.join(cluster_path, "metadata.json")
+    with open(metadata_file, "w+") as f:
+        json.dump(cluster_info, f)
+
+
+def wait_for_rosa_input(spawn, acceptable_duration=5):
+    """
+    pass
```

Comment:
docstring needs to be fixed

Reply:
The method will be removed. It is not used anywhere.